client/packages.go

```diff
@@ -67,8 +67,30 @@ type Snap struct {
 }
 
 type AppInfo struct {
-	Name   string `json:"name"`
-	Daemon string `json:"daemon"`
+	Name         string `json:"name"`
+	DesktopFile  string `json:"desktop-file,omitempty"`
+	*ServiceInfo `json:",omitempty"`
```

Comment:
the empty name means "do the default thing", which for embedded structs means to embed the json (otherwise you could have the json be a subdocument).

This is done in this slightly awkward way so apps don't have all the daemon info empty, but services still have the nice `"active": false` which would be pruned with a simple `omitempty` on the field value.